File: 1894-MergeStringsAlternately/1894-MergeStringsAlternately.cpp
// Last updated: 19/2/2026, 9:13:35 am
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string merged;
        int maxl=max(word1.length(),word2.length());
        for(int i=0;i<maxl;i++){
            if(i<word1.length()){
                merged+=word1[i];
            }
            if(i<word2.length()){
                merged+=word2[i];
            }
        }
        return merged;
        
    }
};

File: 1894-MergeStringsAlternately/1894-MergeStringsAlternately.py
# Last updated: 19/2/2026, 9:13:40 am
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        n, m = len(word1), len(word2)
        res = []
        i = j = 0
        while i < n or j < m:
            if i < n:
                res.append(word1[i])
            if j < m:
                res.append(word2[j])
            i += 1
            j += 1
        return "".join(res)